rhcos.sh

```diff
@@ -23,3 +23,54 @@ export MACHINE_OS_BOOTSTRAP_IMAGE_SHA256=${MACHINE_OS_BOOTSTRAP_IMAGE_SHA256:-${
 # https://github.com/openshift/installer/issues/2845
 export MACHINE_OS_INSTALLER_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256=$(echo "${MACHINE_OS_IMAGE_JSON}" | jq -r '.images.qemu["uncompressed-sha256"]')
 export MACHINE_OS_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256=${MACHINE_OS_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256:-${MACHINE_OS_INSTALLER_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256}}
+
+# Temporary: Use sha256 sums from local storage if present, to make sure
+# we use what was calculated after editing the bootstrap image.
+temporary_sha256_hack() {
+    BOOTSTRAP_IMAGE_FILE="${IRONIC_DATA_DIR}/html/images/${MACHINE_OS_BOOTSTRAP_IMAGE_NAME}"
+    BOOTSTRAP_IMAGE_SHA256_FILE="${BOOTSTRAP_IMAGE_FILE}.sha256sum"
+    UNCOMPRESSED_BOOTSTRAP_IMAGE_FILE="$(echo "${BOOTSTRAP_IMAGE_FILE}" | sed -e 's/\.gz//')"
+    UNCOMPRESSED_BOOTSTRAP_IMAGE_SHA256_FILE="${UNCOMPRESSED_BOOTSTRAP_IMAGE_FILE}.sha256sum"
+
+    if [ ! -f "${UNCOMPRESSED_BOOTSTRAP_IMAGE_SHA256_FILE}" ]; then
+        return
+    fi
+
+    export MACHINE_OS_BOOTSTRAP_IMAGE_SHA256=$(cat "${BOOTSTRAP_IMAGE_SHA256_FILE}" | awk '{print $1}')
+    export MACHINE_OS_BOOTSTRAP_IMAGE_UNCOMPRESSED_SHA256=$(cat "${UNCOMPRESSED_BOOTSTRAP_IMAGE_SHA256_FILE}" | awk '{print $1}')
+}
+temporary_sha256_hack
+
+hack_rhcos_bootstrap_image() {
+    if [[ -n "${USE_IPV4}" ]]; then
+        # Don't hack the image if using IPv4
+        return
+    fi
+
+    IMAGE_FILE="$1"
+    IMAGE_SHA256_FILE="${IMAGE_FILE}.sha256sum"
+    UNCOMPRESSED_IMAGE_FILE="$(echo "${IMAGE_FILE}" | sed -e 's/\.gz//')"
+    UNCOMPRESSED_IMAGE_SHA256_FILE="${UNCOMPRESSED_IMAGE_FILE}.sha256sum"
+
+    pushd $(dirname "${IMAGE_FILE}")
+
+    gunzip "${IMAGE_FILE}"
+    set +e
+    IP_PARAM=$(virt-cat -a "${UNCOMPRESSED_IMAGE_FILE}" -m /dev/sda1 /grub2/grub.cfg | grep "ip=ens3:dhcp6")
+    set -e
+    if [[ -n "${IP_PARAM}" ]] ; then
+        # Image already successfully hacked
+        gzip "${UNCOMPRESSED_IMAGE_FILE}"
+        popd
+        return
+    fi
+
+    # Fix the ip= kernel command line to work for IPv6.  This is temporary.
+    virt-edit -a "${UNCOMPRESSED_IMAGE_FILE}" -m /dev/sda1 -e "s/ip=dhcp/ip=ens3:dhcp6/g" /grub2/grub.cfg
```

Comment:
https://github.com/openshift/installer/commit/9ed508a3a86da759f68c1c55d43b48593c3e2ca4 landed with the fix for https://bugzilla.redhat.com/show_bug.cgi?id=1803926 so we may be able to remove this now (and I'm not sure the virt-edit will work with the new defaults)

Reply:
If the virt-edit needs to stay we may need to figure out a way to edit the image and take a copy without changing the original image(or do caching differently), this image edit is changing the file in place which is one of the few directories that are persisted between jobs.

Reply:
I tested locally and I think the image hack can now be removed for 4.5 builds - I guess we may need to manually purge any modified images though?

Reply:
I disabled the hack for 4.5.

Good point about editing in-place and that breaking the cache in CI!  Now that this doesn't run for 4.5 though, this CI system isn't going to run it anymore ...

Reply:
Ack, I'll purge the image cache in CI and we should be ok for future patches.